autoload/lsp/utils/text_edit.vim

```diff
@@ -80,6 +82,22 @@ function! lsp#utils#text_edit#apply_text_edits(uri, text_edits) abort
 
         let l:i = l:merged_text_edit['end_index']
     endwhile
+
+    call setloclist(0, reverse(l:loc_list), 'a')
+endfunction
+
+function! s:build_loclist_item(uri, text_edit) abort
+    let l:path = lsp#utils#uri_to_path(a:uri)
+    let l:bufnr = bufnr(l:path)
+    let l:lnum = a:text_edit['range']['start']['line'] + 1
+    let l:col = a:text_edit['range']['start']['character'] + 1
```

Comment:
These off-by-one creeps me out a bit, I assume it's due to a difference in how vim and LSP indexing works.

Reply:
you are right. this is vim and lsp indexing difference.